pkg/network/ovn_kubernetes.go

```diff
@@ -197,13 +202,31 @@ func renderOVNKubernetes(conf *operv1.NetworkSpec, bootstrapResult *bootstrap.Bo
 	objs = append(objs, manifests...)
 
 	dpuMode := bootstrapResult.OVN.OVNKubernetesConfig.DPUMode
-	if dpuMode != OVN_DPU_MODE_FULL {
+	if dpuMode == OVN_DPU_MODE_DPU_HOST {
 		data.Data["OVN_DPU_MODE"] = dpuMode
 		manifests, err = render.RenderTemplate(filepath.Join(manifestDir, "network/ovn-kubernetes/ovnkube-node.yaml"), &data)
 		if err != nil {
 			return nil, errors.Wrap(err, "failed to render manifests")
 		}
 		objs = append(objs, manifests...)
+	} else if dpuMode == OVN_DPU_MODE_DPU {
+		// "OVN_DPU_MODE" not set when render.RenderDir() called above,
+		// so render just the error-cni.yaml with "OVN_DPU_MODE" set.
+		data.Data["OVN_DPU_MODE"] = dpuMode
+		manifests, err = render.RenderTemplate(filepath.Join(manifestDir, "network/ovn-kubernetes/error-cni.yaml"), &data)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to render manifests")
+		}
+		objs = append(objs, manifests...)
+
+		// Run KubeProxy on DPU
+		if conf.DeployKubeProxy == nil {
+			v := true
+			conf.DeployKubeProxy = &v
+		} else {
+			*conf.DeployKubeProxy = true
+		}
+		fillKubeProxyDefaults(conf, nil)
```

Comment:
This relies on the fact that `renderStandaloneKubeProxy` happens to get called after `renderDefaultNetwork`... you should set the kube proxy config from `fillOVNKubernetesDefaults`.

Reply:
@danwinship When I originally started coding this up, I started coding it up in `fillKubeProxyDefaults()`. I moved it to it's current location because `FillDefaults()` doesn't have `bootstrapResult` computed yet. None of the defaults are derived from configuration data. I can rearrange the logic where `FillDefaults()` is called, the `bootstrapResult` is generated right after it. Just wasn't sure if that was better or not. If I am doing that, should it go into `fillKubeProxyDefaults()` or `fillDefaultNetworkDefaults()`?

`FillDefaults()` -> https://github.com/openshift/cluster-network-operator/blob/master/pkg/network/render.go#L183
 Where `FillDefaults()` is called -> https://github.com/openshift/cluster-network-operator/blob/master/pkg/controller/operconfig/operconfig_controller.go#L173

Reply:
oh... yeah, we can't change `conf` based on `bootstrapResult`... The bootstrap data is only allowed to affect the rendering output. And actually, `fillKubeProxyDefaults` won't have filled in all of the defaults if `conf.DeployKubeProxy` had been `false` when it was called, so even the way the code is now, it's not getting entirely the right answer.

hm... and I don't really like the changes to `kube_proxy.go` either... it shouldn't be looking in someone else's bootstrap data... the ovn-kube rendering code ought to be able to tell the kube-proxy-rendering code what selector it wants to use. Somehow.

Really though, the answer here is just that we've hit the breaking point of our current abstractions and we need to redesign it a little... but not now.

OK, so eventually the dpu mode will actually be in the network config, right? So we'd have access to it at `fillOVNKubernetesDefaults` time... so I think the best answer that doesn't involve totally rewriting CNO rendering is to move the dpu-configmap-reading code from `bootstrapOVN` into a separate helper function, and then have `fillOVNKubernetesDefaults` and `bootstrapOVN` both call that helper function, and so then we can adjust the defaults based on the dpu mode.

Reply:
Per Discussion with @danwinship, leaving as is. Not ideal, but we are limited because the DPU support is DevPreview and we don't want to change the API until the feature is more stable. Enabling KubeProxy should be done in KubeProxy logic, but when setting up defaults, the code doesn't have access to the DPU Mode field. Currently it is stored in `bootstrapResult.OVN.OVNKubernetesConfig`, but longer term it will be stored in conf (operv1.NetworkSpec), which `FillDefaults` does have access.

Added comments with the tag `DPU_DEV_PREVIEW` in:
* rendor.go, where it is assumed that `renderStandaloneKubeProxy()` is called after `renderDefaultNetwork()`
* kube_proxy.go where it is accessing `bootstrapResult.OVN.OVNKubernetesConfig.DPUMode`
* ovn_kubernetes.go where it enables KubeProxy, which should eventually be moved to KubeProxy logic.